riscv-osps.adoc

```diff
@@ -0,0 +1,302 @@
+implemented on any given platform; if such a requirement
+could cause the platform to behave differently when not implemented,
+a fallback position is specified that must be implemented.
+
+Over time, it is expected that each use case will see a series of
+revisions.  Under ideal circumstances, a platform will be compliant
+with the most recent version of any use case specification.  When a
+platform indicates it is compliant with the OSPS, it should specify
+a use case *and* a specific revision of that use case.
+
+// terms and definitions
+include::terms.adoc[]
+
+## Use Cases
+We define these use cases:
+
+* General purpose computing, where the hardware and software of a system
+are expected to change over time, and configurations of hardware and software
+can be highly variable.
+* Appliance computing, where the hardware configuration is essentially
+fixed, but software is expected to change.
+* Special purpose computing, where hardware and software typically do not
+change once shipped from the factory.
+
+### Common Requirements
+This section is currently a placeholder.
+
+NOTE: As requirements for individual use cases are uncovered, it is
+expected that some will end up being common to all cases.  If so, they
+should be moved to this section.
+
+#### Revision History
+Common Requirements Current Revision: *0* [September 2020]
+
+### General Purpose Computing
+This use case sets out the requirements for what is commonly known
+as _general purpose computing_.  For platforms of this type, it is expected
+that the user will be able to add or remove cards such as PCIe or SATA, 
+add or remove devices such as disks, monitors or graphics cards, and will
+be able to choose from one or more available operating systems.  Commercial
+desktop, laptop or server systems are primary examples.
+
+Operating systems for this use case include at a minimum the members
+of the Linux families (Fedora, Debian, Red Hat, SUSE, and similar).  If
+a platform is compliant with the requirements of this use case, it should
+be possible to easily install and maintain any of the Linux distributions
+without modification.
+
+It should also be possible to use any of the other Unix-based operating
+systems and distributions such as those in the BSD family, as well as
+commercial operating systems such as Microsoft Windows.
+
+The stress here is on the idea of _general_ computing, and hence most of
+the readily available operating systems designed for that environment.  The
+goal is to be able to seamlessly replace existing systems based on other
+ISAs.
+
+#### Revision History
+General Purpose Computing Current Revision: *0* [September 2020]
+
+#### RISC-V ISA Requirements
+##### Machine Mode
+##### Hypervisor Mode
+##### Supervisor Mode
+. Supervisor-mode environments must implement at least version 20190608
+  of the RISC-V Instruction Set Manual Volume II: Privileged
+  Specification <<ISAU>>.
+
+. Supervisor-mode environments must implement RV64GC.
```

Comment:
I'm strongly in favor of being opinionated for the "General Purpose Computing" use case if it can simplify things for everyone. If we want to have a separate "Low Cost General Purpose Computing" use case that seems fine, but 64-bit CPUs with floating point have been basically ubiquitous in laptops and servers for many years.

Reply:
The intent is to start with 64-bit.  If there's interest in 32-bit, we can either get to it later or spin off a subcommittee to handle it.